src/model/utils/deletecontent.js

```diff
@@ -139,6 +139,11 @@ function mergeBranches( writer, startPos, endPos ) {
 	startPos = Position.createAfter( startParent );
 	endPos = Position.createBefore( endParent );
 
+	// One more check if both positions ended up in the same parent. See: https://github.com/ckeditor/ckeditor5/issues/1265.
+	if ( endParent == startPos.parent ) {
```

Comment:
This code should land in `checkCanBeMerged()`. I'm also not sure if this is the right check. See the comment above `checkCanBeMerged()`:

```js

// Check if parents of two positions can be merged by checking if there are no limit/object
// boundaries between those two positions.
//
// E.g. in <bQ><p>x[]</p></bQ><widget><caption>{}</caption></widget>
// we'll check <p>, <bQ>, <widget> and <caption>.
// Usually, widget and caption are marked as objects/limits in the schema, so in this case merging will be blocked.
```

This is a more general implementation – it's based on checking whether there's no limit elements involved. We know that the case we're currently solving requires that everything happens inside a limit element. I can also see that the condition you wrote is true when `endParent` is a `tableCell` (which is a limit element). That makes me suspicious whether the check should be about `endParent` being equal to `startPos.parent`.

Reply:
I changed my solution. Now the merge is blocking if an only single element is between positions (`leftPos` and `rightPos`). From my point of view, such merge does not have any sense because there is nothing to merge.